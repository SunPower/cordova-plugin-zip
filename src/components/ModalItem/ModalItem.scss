@import "../../styles/variables";

.modal-item {
  box-shadow: 4px 4px 4px $gray-shadow-default;
  background: $white;
  color: $black;
  border: 2px solid $gray-border-default;
  padding: 2em 0;

  &.fit-screen {
    flex-grow: 1;
  }

  .left {
    text-align: left;
    padding: 0 2em 2em 2em;
  }

  .right {
    text-align: right;
    padding: 0 2em 2em 2em;
  }

  .center {
    padding: 0 0 2em;
    letter-spacing: $letter-spacing-big;
  }

  .subtitle {
    color: $gray-text-alt2;
    font-weight: $regular-text;
    border-bottom: 2px dashed $gray-border-default;
  }

  .title {
    letter-spacing: $letter-spacing-big;
  }

  .modal-separator {
    width: 50px;
    border-bottom: 1px solid $orange-default;
    margin: 0 auto 20px;
  }
}
