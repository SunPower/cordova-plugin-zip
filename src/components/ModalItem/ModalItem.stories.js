import React from 'react'

import { storiesOf } from '@storybook/react'

import ModalItem from './index'

storiesOf('ModalItem', module).add('Simple', () => <ModalItem />)
