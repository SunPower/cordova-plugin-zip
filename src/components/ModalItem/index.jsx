import React from 'react'
import clsx from 'clsx'
import './ModalItem.scss'

function ModalItem({
  overtitle = null,
  overtitleAlign = null,
  className = '',
  paddingLeft = 'pl-10',
  paddingRight = 'pr-10',
  title,
  children
}) {
  const classes = clsx('modal-item', 'content', className)
  const headerClasses = clsx(
    'subtitle',
    'is-spaced',
    'is-6',
    overtitleAlign || 'has-text-centered center'
  )
  const contentClasses = clsx('content', paddingLeft, paddingRight)
  return (
    <div className={classes}>
      {overtitle && <h3 className={headerClasses}>{overtitle}</h3>}
      {title && (
        <React.Fragment>
          <h2 className="title is-uppercase has-text-centered pl-20 pr-20 is-6">
            {title}
          </h2>
          <div className="modal-separator"></div>
        </React.Fragment>
      )}
      <div className={contentClasses}>{children}</div>
    </div>
  )
}

export default ModalItem
