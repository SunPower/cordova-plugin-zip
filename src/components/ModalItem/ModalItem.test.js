import React from 'react'
import { shallow } from 'enzyme'
import ModalItem from '.'

describe('ModalItem component', () => {
  it('renders without crashing', () => {
    const component = shallow(<ModalItem />)
    expect(component).toMatchSnapshot()
  })
})
