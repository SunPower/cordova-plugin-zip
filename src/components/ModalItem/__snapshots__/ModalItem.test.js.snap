// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalItem component renders without crashing 1`] = `
<div
  className="modal-item content"
>
  <div
    className="content pl-10 pr-10"
  />
</div>
`;
